docs/root/configuration/advanced/well_known_filter_state.rst

```diff
@@ -12,6 +12,20 @@ The following list of filter state objects are consumed by Envoy extensions:
 
    * - **Filter state key**
      - **Purpose**
+   * - ``envoy.network.upstream_server_name``
```

Comment:
nit: I think `server_name` is ambiguous as it may refer to a host name as well. Using the `upstream_server_name_indication` may disambiguate.

Reply:
We cannot change the filter state key without breaking someone. I'm hoping by having this table we can make a better naming decision next time we add a new filter state object.